diff --git a/packages/dev/src/chassis.ts b/packages/dev/src/chassis.ts
index 317a5c24..32c44908 100644
--- a/packages/dev/src/chassis.ts
+++ b/packages/dev/src/chassis.ts
@@ -283,6 +283,9 @@ export default async (
         const mods = env.moduleGraph.getModulesByFile(file);
         if (mods?.size) {
           try {
+            for (const mod of mods) {
+              env.moduleGraph.invalidateModule(mod);
+            }
             await service.teardown?.();
             await close();
             service = await loadService();
